cmd/iam_api_key_delete.go

```diff
@@ -71,15 +53,15 @@ func (c *iamAPIKeyDeleteCmd) cmdRun(_ *cobra.Command, _ []string) error {
 		}
 	}
 
-	var err error
-	decorateAsyncOperation(fmt.Sprintf("Deleting API Key %s...", c.APIKey), func() {
-		err = globalstate.EgoscaleClient.DeleteAPIKey(ctx, zone, &egoscale.APIKey{Key: &c.APIKey})
-	})
-	if err != nil {
-		return err
-	}
+	return decorateAsyncOperations(fmt.Sprintf("Deleting API Key %s...", c.APIKey), func() error {
+		op, err := client.DeleteAPIKey(ctx, apiKey.Key)
+		if err != nil {
+			return fmt.Errorf("exoscale: error while deleting IAM API Key: %w", err)
+		}
 
-	return nil
+		_, err = client.Wait(ctx, op, v3.OperationStateSuccess)
+		return err
```

Comment:
Thanks for refactoring it, like the line above as you wrapped the error, I think you can do the same here :) 
it can apply to the rest of occurrences.

Reply:
still this one before merging